File: problem=016/Solution.js
// I had to use 12.x version of NodeJS for this.

const solve = exponent =>
	(2n ** BigInt(exponent))
		.toString()
		.split("")
		.map(x => Number.parseInt(x, 10))
		.reduce((s, i) => s + i, 0);

console.log(solve(15));
console.log(solve(1000));

// If this solution helped you, please star the repo!

File: problem=016/Solution.java
import java.math.BigInteger;

final class Solution {

	private static int solve(int exponent) {
		final char chars[] = BigInteger.TWO.pow(exponent).toString(10).toCharArray();
		int sum = 0;
		for (final char c : chars) {
			sum += c - '0';
		}
		return sum;
	}

	public static void main(String[] args) throws Exception {
		System.out.println(solve(15));
		System.out.println(solve(1000));
	}
}

// If this solution helped you, please star the repo!